runtime/src/bank.rs

```diff
@@ -1260,6 +1263,7 @@ impl Bank {
         bank.update_rent();
         bank.update_epoch_schedule();
         bank.update_recent_blockhashes();
+        bank.fill_sysvar_cache();
```

Comment:
Here, we are saving the accounts to the db in the "update" calls, and then immediately reloading them to populate the cache.  Each of these operations requires an accountsdb lock.  Can we instead combine these into a more efficient operation?

Reply:
`fill_sysvar_cache()` only tries to load them if they are not found in the cache already. So the fill happens after all sysvar updates deliberately.

`if !sysvar_cache.iter().any(|(key, _data)| key == id) {`

So the inefficiency I can see here is:
- The linear search in the `sysvar_cache`
- And that each `update_sysvar_account()` call reacquires the locks

Reply:
Yeah, that locking...

Reply:
I agree that it probably can be optimized. But that is not a change specific to this PR. So same question as above:

> Can we open a separate issue for that and handle it in a different PR?

Reply:
It is specific because the cache is adding a lot ore locks, but yeah, we can optimize that out in another pr